executor/linux/build.go

```diff
@@ -554,6 +665,21 @@ func (c *client) DestroyBuild(ctx context.Context) error {
 		}
 	}
 
+	// destroy the secrets for the pipeline
+	for _, s := range c.pipeline.Secrets {
+		// skip over non-plugin secrets
+		if s.Origin.Empty() {
+			continue
+		}
+
+		c.logger.Infof("destroying %s secret", s.Name)
+		// destroy the service
```

Comment:
I think this comment needs to be updated 👍

```suggestion
		// destroy the secret
```